diff --git a/partitionmanager.cpp b/partitionmanager.cpp
index 31f7882..3fd80e6 100755
--- a/partitionmanager.cpp
+++ b/partitionmanager.cpp
@@ -2102,9 +2102,11 @@ void TWPartitionManager::Post_Decrypt(const string& Block_Device) {
 		}
 
 		if (dat->Has_Data_Media && TWFunc::Path_Exists("/data/media/0")) {
-			dat->Storage_Path = "/data/media/0";
+			//dat->Storage_Path = "/data/media/0";
+			dat->Storage_Path = "/sdcard/0";
 		} else {
-			dat->Storage_Path = "/data/media";
+			//dat->Storage_Path = "/data/media";
+			dat->Storage_Path = "/sdcard";
 		}
 		dat->Symlink_Path = dat->Storage_Path;
 		DataManager::SetValue("tw_storage_path", dat->Symlink_Path);
